### ISSUE TITLE
add bindings for CoreWLAN

### ISSUE THREAD

**Issue by SirWindfield** [servo/core-foundation-rs#338]
Not sure if this belongs here. It would be awesome to have bindings to the CoreWLAN interface which provides functionality for WLAN and AirDrop. 

**Documentation:** https://developer.apple.com/documentation/corewlan
**Since:**: macOS 10.6+

I do have access to a MacBook. If you guys could provide me a small how-to I can try to implement it myself and create a PR.

**Comment by jrmuizel** [comment]
I'd recommend doing this in a separate crate that depends on core-foundation. Perhaps take a look at https://github.com/kornelski/rust-security-framework for inspiration.

**Comment by SirWindfield** [comment]
Thank you very much! This question applies to any C interop I guess, but what exactly is the difference between the `crate` and `crate_sys` types? E.g. I checked out the project that you linked in your response. I just thought that `_sys`d are raw bindings whereas the normal `crate` would provide "rust-ified" types. Why would the non-sys crate ever want to import from `core_foundation_sys`? Couldn't it always use `core_foundation` instead?

**Comment by jrmuizel** [comment]
Yes, I believe that's generally the approach to take. There may be situations where importing core_foundation_sys makes sense but none jump to mind.

**Comment by SirWindfield** [comment]
Thanks for the explanation :)
